src/Raven.Server/Documents/PeriodicBackup/Restore/AbstractRestoreBackupTask.cs

```diff
@@ -581,7 +581,19 @@ protected virtual void ConfigureSettingsForSmugglerRestore(DocumentDatabase data
                     databaseRecord.DataArchival = smugglerDatabaseRecord.DataArchival;
                     databaseRecord.QueueSinks = smugglerDatabaseRecord.QueueSinks;
                     databaseRecord.SupportedFeatures = smugglerDatabaseRecord.SupportedFeatures;
+
+                    //Enforce license limitations on the restored database record.
+                    using (ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+                    using (context.OpenReadTransaction())
+                    {
+                        var items = context.Transaction.InnerTransaction.OpenTable(ClusterStateMachine.ItemsSchema, ClusterStateMachine.Items);
+                        foreach (var command in ClusterStateMachine._licenseLimitsCommandsForCreateDatabase)
```

Comment:
I do not like that, introduce a method AssertAllLicenseLimits in the state machine